lib/galaxy/workflow/extract.py

```diff
@@ -161,7 +162,20 @@ class FakeJob:
 
     def __init__(self, dataset):
         self.is_fake = True
-        self.id = "fake_%s" % dataset.id
+        self.id = f"fake_{dataset.id}"
+        self.name = self._guess_name_from_dataset(dataset)
+
+    def _guess_name_from_dataset(self, dataset) -> Optional[str]:
+        """Tries to guess the name of the fake job from the dataset associations."""
+        name = None
+        try:
+            if dataset.copied_from_history_dataset_association:
+                name = "Import from History"
+            elif dataset.copied_from_library_dataset_dataset_association:
+                name = "Import from Library"
+            return name
+        except Exception:
```

Comment:
Is there a specific type of error that's expected here? If so, should we log it? If not, and since there's no error handling involved, do we need to wrap this in a try clause? `None` is returned implicitly, so it could be as simple as if [case 1] return x; elif [case 2] return y.

Reply:
Thank you @ic4f for checking.
> Is there a specific type of error that's expected here?

Probably an `AttributeError` but it is only expected in the unit tests.

> do we need to wrap this in a try clause?

This was an attempt to say, whatever happens here... just return the default value None. I definitely don't like the general catch.

In any case, it was a 'not very elegant' attempt to pass [this unit test](https://github.com/galaxyproject/galaxy/blob/6e59f9b919bd88efbbf894cb603c13ec5118171e/test/unit/workflows/test_extract_summary.py#L48) that was failing because the attribute `copied_from_library_dataset_dataset_association` was not defined in [MockHda here](https://github.com/galaxyproject/galaxy/blob/6e59f9b919bd88efbbf894cb603c13ec5118171e/test/unit/workflows/test_extract_summary.py#L129).
My first thought was to add the attribute to the MockHda (like `copied_from_history_dataset_collection_association` which is already there), but then I thought that this will require maintaining the mock with every other attribute that may be used to guess the name so I went for this ugly solution... 
Do you think would be better just to add the attribute to the mock and remove the try? Any suggestions are welcome :)

Reply:
Oh, I see. So, in my opinion, your first approach was spot on - defining it in the mock next to `copied_from_history_dataset_collection_association` makes perfect sense to me. (maintaining mocks shouldn't be an issue). 

You could also set that attribute on the mock *in the test itself*. In fact, that way you can add a separate unit test that would directly test the new functionality you've added! (you could create 3 HDAs in the same test method, mocking each one differently (you might need another `MockHDA` as the parent (`hda.copied_from_history_dataset_association = MockHda(job=UNDEFINED_JOB)`). And then test the 3 guessed names.

Reply:
Thanks for the advice! I have removed the catch and added a unit test for the `FakeJob` name guessing. I am happier now 😄

Reply:
Thank you - looks great! And very good to have a unit test covering this.